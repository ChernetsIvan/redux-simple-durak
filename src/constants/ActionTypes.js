export const START_GAME_CLICKED = 'START_GAME_CLICKED';
export const BEGIN_GAME_CLICKED = 'BEGIN_GAME_CLICKED';
