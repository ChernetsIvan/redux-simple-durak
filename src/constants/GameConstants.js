export const cardRanks = [
  { text: "6", cardValue: 1 },
  { text: "7", cardValue: 2 },
  { text: "8", cardValue: 3 },
  { text: "9", cardValue: 4 },
  { text: "10", cardValue: 5 },
  { text: "В", cardValue: 6 },
  { text: "Д", cardValue: 7 },
  { text: "К", cardValue: 8 },
  { text: "Т", cardValue: 9 }
];

export const cardSuits = [
  { suit: "Ч" },
  { suit: "К" },
  { suit: "П" },
  { suit: "Б" }
];

export const statusMessages = {
  PlayerAttack: "Ваш ход!",
  PlayerDefence: "Защищайтесь!",
  PlayerDiscard: "Беру! Подбрасывать будете?",
  PlayerLose: "Вы проиграли! Сыграем ещё раз ? :)",
  PlayerWin: "Вы выиграли! Поздравляю :)",
  GameResultDraw: "Ничья!"
};
