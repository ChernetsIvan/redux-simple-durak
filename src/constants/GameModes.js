export const PlayerAttack = "PlayerAttack";
export const PlayerDefence = "PlayerDefence";
export const PlayerDiscard = "PlayerDiscard"; //AI забирает: игрок подбрасывает ему карты
export const AiAttack = "AiAttack";
export const AiDefence = "AiDefence";
export const AiDiscard = "AiDiscard"; //Игрок забирает: AI подбрасывает ему карты

export const PlayerWin = "PlayerWin";
export const AIWin = "AIWin";
export const TheDraw = "TheDraw"; //Ничья