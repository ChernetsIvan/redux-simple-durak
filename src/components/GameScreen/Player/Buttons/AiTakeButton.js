import React from "react";
import PropTypes from "prop-types";

import * as gameModes from "./../../../../constants/GameModes";

const AiTakeButton = ({ gameMode, onAiTakeClick }) => {
  let dash = <span>&mdash;</span>;
  return gameMode === gameModes.PlayerDiscard ? (
    <button className="btn btn-success" onClick={onAiTakeClick}>
      {dash} Подбросить нечего, забирай!
    </button>
  ) : null;
};

export default AiTakeButton;

AiTakeButton.propTypes = {
  gameMode: PropTypes.string.isRequired,
  onAiTakeClick: PropTypes.func.isRequired
};
