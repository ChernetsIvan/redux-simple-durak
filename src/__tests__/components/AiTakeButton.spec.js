import React from "react";
import renderer from "react-test-renderer";

import AiTakeButton from "./../../components/GameScreen/Player/Buttons/AiTakeButton";
import * as gameModes from "./../../constants/GameModes";

describe("PlayerTakeButton snapshot testing", () => {
  const emptyFunc = () => {};

  test("will render itself markup", () => {
    const component = renderer.create(
      <AiTakeButton
        gameMode={gameModes.PlayerDiscard}
        onAiTakeClick={emptyFunc}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("nothing render", () => {
    const component = renderer.create(
      <AiTakeButton
        gameMode={gameModes.PlayerAttack}
        onAiTakeClick={emptyFunc}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
