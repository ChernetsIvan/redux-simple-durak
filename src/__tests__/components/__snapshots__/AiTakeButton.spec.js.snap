// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerTakeButton snapshot testing nothing render 1`] = `null`;

exports[`PlayerTakeButton snapshot testing will render itself markup 1`] = `
<button
  className="btn btn-success"
  onClick={[Function]}
>
  <span>
    —
  </span>
   Подбросить нечего, забирай!
</button>
`;
